src/components/lyne-toast/lyne-toast.custom.d.ts

```diff
@@ -0,0 +1,75 @@
+import { AnimationBuilder } from '../../global/core/components/animations/animation-interface';
+
+export interface InterfaceLyneToastAttributes {
+  ariaLivePoliteness: 'off' | 'assertive' | 'polite';
+  horizontalPosition: 'left' | 'center' | 'right' | 'start' | 'end';
+  verticalPosition: 'top' | 'bottom';
+}
+
+type InterfaceToastType = 'link' | 'action' | 'icon';
+
+/**
+ * Common interface for toast action.
+ */
+interface InterfaceToastCommonAction {
+  type: InterfaceToastType;
+  role: 'cancel' | string;
```

Comment:
You are totally right; 'cancel' is not present in the MDN role list. 
However, Ionic uses it (I think) for two reason (see usages of https://github.com/ionic-team/ionic-framework/blob/1641dbc52a07567953bc2f07c4bc828ed95a5185/core/src/utils/overlays.ts#L570):
- since they could have multiple button in alert/toast/etc, it's used to connect the right handler to the button with role 'cancel';
- when the alert/toast/ecc is dismissed via cancel button, the role 'cancel' is emitted in the will/didDismiss events, so maybe consumers can bind their own logic (eg. toast autoclose emits a 'timeout' value instead; see also https://62a9b9a402dd4701997253a8--lyne-components-storybook.netlify.app/?path=/story/components-overlay-lyne-toast--icon-and-close-icon-action-with-event-listeners ).

Reply:
Ok, thanks for your clarification, so we leave it for now :-)